assets/src/edit-story/components/transform/transformProvider.js

```diff
@@ -37,16 +39,39 @@ function TransformProvider({ children }) {
     };
   }, []);
 
-  const pushTransform = useCallback((id, transform) => {
-    const handlerListMap = transformHandlersRef.current;
-    const handlerList = handlerListMap[id];
-    if (handlerList) {
-      handlerList.forEach((handler) => handler(transform));
-    }
-  }, []);
+  const pushTransform = useCallback(
+    (id, transform) => {
+      const handlerListMap = transformHandlersRef.current;
+      const lastTransforms = lastTransformsRef.current;
+      const handlerList = handlerListMap[id];
+
+      lastTransforms[id] = transform;
+
+      if (handlerList) {
+        handlerList.forEach((handler) => handler(transform));
+      }
+
+      if (!isAnythingTransforming && transform !== null) {
+        setIsAnythingTransforming(true);
+      }
+
+      if (isAnythingTransforming && transform === null) {
+        const allTransformsDone = Object.values(lastTransforms).every(
+          (v) => v === null
+        );
+        if (allTransformsDone) {
+          lastTransformsRef.current = {};
+          setIsAnythingTransforming(false);
+        }
+      }
+    },
+    [isAnythingTransforming]
```

Comment:
This will make this a bit slower and we want `pushTransform` to be very fast. I'm looking at the code, however, it doesn't need to depend on the `isAnythingTransforming`. `setIsAnythingTransforming()` will do the right thing if the value hasn't actually changed.